include/RED4ext/SharedSpinLock.hpp

```diff
@@ -4,18 +4,18 @@
 
 namespace RED4ext
 {
-struct SharedMutex
+struct SharedSpinLock
```

Comment:
Imo, we should add an alias to this and mark `SharedMutex` as deprecated.

Reply:
I've added an alias with deprecation